mir_eval/sonify.py

```diff
@@ -103,11 +103,18 @@ def time_frequency(gram, frequencies, times, fs, function=np.sin, length=None,
     if length is None:
         length = int(times[-1, 1] * fs)
 
-    times, _ = util.adjust_intervals(times, t_max=length)
-
-    # Truncate times so that the shape matches gram
-    n_times = gram.shape[1]
-    times = times[:n_times]
+    last_time_in_secs = float(length) / fs
+    times, _ = util.adjust_intervals(times, t_max=last_time_in_secs)
+
+    # Truncate times so that the shape matches gram. However if the time boundaries were converted
+    # to intervals, then the number of times will be reduced by one, so we only truncate
+    # if the gram is smaller.
```

Comment:
This could be more compact:

```python
n_times = min(gram.shape[1], times.shape[0])
times = times[:n_times]
```

Reply:
As the diagnostic code displays, and the comment explains, the truncation of `times`, after adjusting the intervals, should only be done if the `times` length exceeds the `gram` length. This would lead to an unnecessary truncation operation if the `gram` length equals or exceeds the `times` length. Whether that truncation operation is performed or becomes a no-op would depend on the implementation of numpy.

Reply:
> This would lead to an unnecessary truncation operation if the `gram` length equals or exceeds the `times` length. Whether that truncation operation is performed or becomes a no-op would depend on the implementation of numpy.

I'm not sure I understand your comment here.  `x[:x.shape[0]]` is well-defined in the numpy API, and would not invoke a copy, nor would it have side-effects at the call site since `gram` is a reference.

In general, this kind of thing should always produce a view instead of a copy, and is essentially free:
```python
In [1]: import numpy as np

In [2]: x = np.arange(10)

In [3]: y = x[:x.shape[0]]

In [4]: y.flags
Out[4]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
```
(note the `OWNDATA` flag being false there.)

Reply:
OK, fair comment, I've updated the code to your suggested improvements.